core/modules/menu/menu.module

```diff
@@ -81,14 +81,24 @@ function menu_menu() {
     'type' => MENU_DEFAULT_LOCAL_TASK,
     'context' => MENU_CONTEXT_PAGE | MENU_CONTEXT_INLINE,
   );
-  $items['admin/structure/menu/manage/%menu/add'] = array(
-    'title' => 'Add link',
-    'page callback' => 'backdrop_get_form',
-    'page arguments' => array('menu_edit_item', 'add', NULL, 4),
-    'access arguments' => array('administer menu'),
-    'type' => MENU_LOCAL_ACTION,
-    'file' => 'menu.admin.inc',
-  );
+  foreach (menu_get_menus() as $name => $menu) {
+    $items['admin/structure/menu/manage/'. $name . '/add'] = array(
+      'title' => $menu . ': Add link',
+      'page callback' => 'backdrop_get_form',
+      'page arguments' => array('menu_edit_item', 'add', NULL, 4),
+      'access arguments' => array('administer menu'),
+      'type' => MENU_LOCAL_ACTION,
+      'file' => 'menu.admin.inc',
+    );
+  }
+  // $items['admin/structure/menu/manage/%menu/add'] = array(
+  //   'title' => 'Add link',
+  //   'page callback' => 'backdrop_get_form',
+  //   'page arguments' => array('menu_edit_item', 'add', NULL, 4),
+  //   'access arguments' => array('administer menu'),
+  //   'type' => MENU_LOCAL_ACTION,
+  //   'file' => 'menu.admin.inc',
+  // );
```

Comment:
This corrects the search results so the menu entries are now *Structure: [menu_name]: Add link*, but it also changes the actual menu items too. So from:

*Structure* -> *Menus* -> *[menu_name]* -> *Add link*

...you now get:

*Structure* -> *Menus* -> *[menu_name]: Add link*

Notice how the "Add link" menu items have now moved one level up. Not good. I though that the idea here was to add the ```[menu_name]:``` part **only** to the search results while leaving the menu items as they are.